gradle/publish.gradle

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Configures publishing of the main java projects
+ */
+
+// Publishing helpers
+ext {
+    configurePom = { project ->
+        { ->
+                name = project.hasProperty('pomName') ? project.getProperty('pomName') : project.name
+                description = project.description
+                url = project.hasProperty('pomURL') ? project.getProperty('pomURL') : 'http://www.mongodb.org'
+                licenses {
+                    license {
+                        name = 'The Apache License, Version 2.0'
+                        url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                    }
+                }
+                scm {
+                    url = 'https://github.com/mongodb/mongo-java-driver'
+                    connection = 'scm:https://github.com/mongodb/mongo-java-driver.git'
+                    developerConnection = 'scm:git@github.com:mongodb/mongo-java-driver.git'
+                }
+                developers {
+                    developer {
+                        name = 'Various'
+                        organization = 'MongoDB'
+                    }
+                }
+            }
+    }
+    configureMavenRepositories = { project ->
+        { ->
+            def snapshotsRepoUrl = 'https://oss.sonatype.org/content/repositories/snapshots/'
+            def releasesRepoUrl = 'https://oss.sonatype.org/service/local/staging/deploy/maven2/'
+            maven {
+                url = project.version.endsWith('SNAPSHOT') ? snapshotsRepoUrl : releasesRepoUrl
+                credentials {
+                    username project.hasProperty('nexusUsername') ? project.getProperty('nexusUsername') : ''
+                    password project.hasProperty('nexusPassword') ? project.getProperty('nexusPassword') : ''
+                }
+            }
+        }
+    }
+    configureSigning = { project ->
+        { ->
+            def isReleaseVersion = !project.version.endsWith('SNAPSHOT')
+            signing {
+                required { isReleaseVersion && gradle.taskGraph.hasTask('uploadArchives') }
```

Comment:
I don't think I understand this.  This seems to indicate that we only sign release versions, but we do also sign snapshots.

Reply:
For some reason I thought we didn't sign the snapshots - I'll remove this. Looks like it only signs the main jar - so will need to review and ensure the javadoc & poms also get signed.